src/Cake.Core.Tests/Unit/IO/NuGet/NuGetPackageAssembliesLocatorTests.cs

```diff
@@ -0,0 +1,139 @@
+using System.Collections.Generic;
+using System.Runtime.Versioning;
+using Cake.Core.Diagnostics;
+using Cake.Core.IO;
+using Cake.Core.IO.NuGet;
+using Cake.Testing;
+using NSubstitute;
+using Xunit;
+
+namespace Cake.Core.Tests.Unit.IO.NuGet
+{
+    public sealed class NuGetPackageAssembliesLocatorTests
+    {
+        public sealed class TheFindAssembliesMethod
```

Comment:
Tests do a lot of in the `Given` step. Maybe refactor out to a fixture and add methods to set the fixture in the right state. Would increase readability of the tests.